compendium/DeclarativeServices/test/bench/CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+include_directories(SYSTEM ${CppMicroServices_SOURCE_DIR}/third_party/boost/include)
+
+
+if (US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG)
+  check_cxx_compiler_flag(-Wno-inconsistent-missing-override HAS_MISSING_OVERRIDE_FLAG)
+  if (HAS_MISSING_OVERRIDE_FLAG)
+    add_compile_options(-Wno-inconsistent-missing-override)
+  endif()
+  check_cxx_compiler_flag(-Wno-deprecated-declarations HAS_MISSING_DEPRECATED_DECLARATIONS_FLAG)
+  if (HAS_MISSING_OVERRIDE_FLAG)
+    add_compile_options(-Wno-deprecated-declarations)
+  endif()
+endif()
+
+if(MSVC)
+  add_compile_definitions(GTEST_HAS_STD_TUPLE_=1)
+  add_compile_definitions(GTEST_HAS_TR1_TUPLE=0)
+  add_compile_definitions(GTEST_LANG_CXX11=1)
+endif()
+
+add_compile_definitions(BOOST_DATE_TIME_NO_LIB)
+add_compile_definitions(BOOST_REGEX_NO_LIB)
+
+
+#-----------------------------------------------------------------------------
+# Add test source files
+#-----------------------------------------------------------------------------
+set(_declarativeservices_benchmark_tests
+  GetDSServiceTest.cpp
+)
+
+include_directories(${PROJECT_BINARY_DIR}/include
+  ${CppMicroServices_SOURCE_DIR}/compendium/CM/include
+  ${CppMicroServices_BINARY_DIR}/compendium/CM/include
+  )
+
+set(_additional_srcs
+  ../TestUtils.cpp
+  ../../src/SCRActivator.cpp
+  )
+
+#-----------------------------------------------------------------------------
+# Build the main test driver executable
+#-----------------------------------------------------------------------------
+
+# Generate a custom "bundle init" file for the test driver executable
+usFunctionGenerateBundleInit(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  OUT _additional_srcs
+  )
+
+usFunctionGetResourceSource(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  OUT _additional_srcs)
+
+add_executable(${us_declarativeservices_bench_test_exe_name}
+  ${_declarativeservices_benchmark_tests}
+  ${_additional_srcs})
+
+if (US_COMPILER_MSVC AND BUILD_SHARED_LIBS)
+  target_compile_options(${us_declarativeservices_bench_test_exe_name} PRIVATE -DGTEST_LINKED_AS_SHARED_LIBRARY)
+endif()
+
+set_property(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  APPEND PROPERTY COMPILE_DEFINITIONS US_BUNDLE_NAME=main
+  )
+
+set_property(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  PROPERTY US_BUNDLE_NAME main
+  )
+
+target_include_directories(${us_declarativeservices_bench_test_exe_name}
+  PRIVATE $<TARGET_PROPERTY:util,INCLUDE_DIRECTORIES>)
+
+# Disable deprecation warnings.
+if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") OR
+    ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") OR
+    ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+  set_source_files_properties(${_declarativeservices_benchmark_tests}
+    PROPERTIES
+    COMPILE_FLAGS -Wno-error=deprecated-declarations
+    COMPILE_FLAGS -Wno-sign-compare
+    )
+endif()
+
+set(_test_bundles
```

Comment:
I'd like to see the set of test bundles trimmed down to only what is needed right now. As we add more, this list can grow at that point.